docs/README.md

```diff
@@ -100,6 +100,49 @@ Creating a pull request (PR) can only be done with on the [GitHub website](https
 
 
+---
+
+### Working together
+
+You might at some point end up working on feature `X` together with another – or more – teammate(s). Dealing with this situation unprepared *will* lead you down a rabbit hole of painful and seemingly endless merge conflicts. 
+
+In an attempt to avoid as many as these merge conflicts, here's what you should **avoid**:
+- Do not, under any circumstances, copy-paste code snippets between students/branches to make your code compile. You should rather ask your teammate to push their changes online, and incorporate them into your work (see below how we suggest you achieve this);
+
+Here's how we suggest you work on feature `X` with two (or more) teammates:
+1. Create a branch that is common between both of you. If feature `X` relies on branch `Y` (that is awaiting reviews, i.e. not merged into `master`), then create your common branch originating from branch `Y`. If it is not related to any unmerged branches, then default to a copy of `master`.
+2. Create a personal branch for each student that is originating from the common branch. Additionally (arguably optionally), you may decide to never push changes directly to the common branch but instead use PRs from either of your personal branches to the common branch.
+3. When you have a self-contained piece of code that works, add it to the common branch (either through PRs or `git merge`). This way, your partner will be able to use it without the need of copy-pastas.
+4. When you need new code that is on the common branch but not yet on your personal branch, simply merge the common branch inside your personal branch.
+5. Once you are all happy and the feature is fully finished (and thus operational on the common branch), simply create a PR from the common branch to branch `Y` if `Y` has not been merged yet or `master` if `Y` has already been merged.
+
+ℹ️ It is desirable to merge `master` into your common branch from time to time in order to avoid said branch from getting too out-of-date with the rest the team.
+
+Here's what it could look like with student Pierluca and student Nicolas working together on the feature `X` in a common branch that is using features from `Y`:
+
+```
+      +----------------------------+
+      |           Master           |
+      +-------------+--------------+
+                    |
+                    |
+      +-------------+--------------+
+      |          branch-y          |
+      +-------------+--------------+
+                    |
+                    |
+      +-------------+--------------+
+      |       dev-common-pn        |
+      +----------------------------+
+           ^ ^                ^
+           | |                | 
+           | |                |
+      +----+-+---+       +----+----+
+      | Pierluca |       | Nicolas |
```

Comment:
I approved but later realized that the naming here doesn't align anymore with what's in the text above 😅